WalletWasabi.Daemon/Global.cs

```diff
@@ -261,8 +271,23 @@ private async Task StartTorProcessManagerAsync(CancellationToken cancellationTok
 			using (BenchmarkLogger.Measure(operationName: "TorProcessManager.Start"))
 			{
 				TorManager = new TorProcessManager(TorSettings);
-				await TorManager.StartAsync(attempts: 3, cancellationToken).ConfigureAwait(false);
+				var (_, torControlClient) = await TorManager.StartAsync(attempts: 3, cancellationToken).ConfigureAwait(false);
 				Logger.LogInfo($"{nameof(TorProcessManager)} is initialized.");
+
+				if (Config is {JsonRpcServerEnabled: true, OnionEnabled: true} && torControlClient is { } nonNullTorControlClient)
+				{
+					var anonymousAccessAllowed = string.IsNullOrEmpty(Config.JsonRpcUser) || string.IsNullOrEmpty(Config.JsonRpcPassword);
+					if (!anonymousAccessAllowed)
+					{
+						var onionServiceId = await nonNullTorControlClient.CreateHiddenServiceAsync(80, 37129, cancellationToken).ConfigureAwait(false);
```

Comment:
This is the very first iteration of this thing and I am not sure the port should  be static, I think it would be better to add that to the config file in the immediate future.

Reply:
Done.